server/src/api/customers.ts

```diff
@@ -74,10 +80,24 @@ customerRouter.post(
     try {
       // Customers are unique by their gpayId, so we will retrieve
       // the customer using their gpay Id.
-      const existingCustomer = await customerService.getCustomerWithGpayId(
+      let existingCustomer = await customerService.getCustomerWithGpayId(
         customerData.gpayId
       );
       if (existingCustomer !== null) {
+        // Update gpayContactNumber if it does not match
```

Comment:
Why do you update the contact number when it does not match with existing customer's existing record? Shouldn't you refuse login because of this mismatch?

Reply:
@xlai20 User might have changed their phone number on GPay between subsequent logins. We have no way for us to know whether the user has changed their phone number for us to react to that change. The only thing we can do is try to call the getPhoneNumber API to retrieve the current phone number of the customer.

So now the workaround is to get customer's phone number when we request a login and update it on the server side if there is a change in phone number, which is what I am implementing in this PR!

I know that this is not the best because this means users can try to manually create this request to change their gpay phone number and break our messages functionality. But because we rely on Spot API to give us the updated phone number, and this Phone Number API is only available on the client side, I have decided to go with implementing it this way.